bpf/bpf_host.c

```diff
@@ -409,11 +409,13 @@ tail_handle_ipv6(struct __ctx_buff *ctx, const bool from_host)
 		if (from_host)
 			invoke_tailcall_if(is_defined(ENABLE_HOST_FIREWALL),
 					   CILIUM_CALL_IPV6_CONT_FROM_HOST,
-					   tail_handle_ipv6_cont_from_host);
+					   tail_handle_ipv6_cont_from_host,
```

Comment:
How about
```
ret = invoke_tailcall_if(is_defined(ENABLE_HOST_FIREWALL),
					   CILIUM_CALL_IPV6_CONT_FROM_HOST,
					   tail_handle_ipv6_cont_from_host);
```
?

Reply:
That would be easier to read, but I couldn't think of a pretty way to make it work for when the macro expands into a tail-call.

Reply:
But then it just doesn't return, so something like
```
#define __invoke_tailcall_if_1(NAME, FUNC)	({			\
		ep_tail_call(ctx, NAME);				\
                DROP_MISSED_TAIL_CALL;				\
})
```
should work

Reply:
yes that should do the trick, thanks :pray: